### ISSUE TITLE
addCharm() takes 3 positional arguments but 4 were given

### ISSUE THREAD

**Issue by reversefold** [conjure-up/conjure-up#1645]
## Report

Thank you for trying **conjure-up**! Before reporting a bug please make sure you've gone through this checklist:

- [x] Is this problem already documented at https://docs.ubuntu.com/conjure-up/en/troubleshoot#common-spell-problems
-- No
- [x] Is conjure-up running inside a virtual machine?
-- AWS
- [x] Is this problem reproducible with `sudo snap refresh conjure-up --edge`?
-- Yes

```
2020-02-24 18:22:03,211 [DEBUG] conjure-up/charmed-kubernetes - events.py:52 - Awaiting DeploymentComplete at conjureup/controllers/juju/deploy/common.py:41 in task _wait_for_applications at conjureup/controllers/juju/deploy/gui.py:82
2020-02-24 18:22:03,266 [DEBUG] conjure-up/charmed-kubernetes - events.py:52 - Setting Error at conjureup/events.py:149
2020-02-24 18:22:03,267 [ERROR] conjure-up/charmed-kubernetes - events.py:161 - Unhandled exception in <Task finished coro=<do_deploy() done, defined at /snap/conjure-up/1058/lib/python3.6/site-packages/conjureup/controllers/juju/deploy/common.py:11> exception=TypeError('addCharm() takes 3 positional arguments but 4 were given',)>
Traceback (most recent call last):
  File "/snap/conjure-up/1058/lib/python3.6/site-packages/conjureup/controllers/juju/deploy/common.py", line 30, in do_deploy
    await app.juju.client.deploy(fn)
  File "/snap/conjure-up/1058/lib/python3.6/site-packages/juju/model.py", line 1308, in deploy
    await handler.execute_plan()
  File "/snap/conjure-up/1058/lib/python3.6/site-packages/juju/model.py", line 1991, in execute_plan
    result = await method(*step.args)
TypeError: addCharm() takes 3 positional arguments but 4 were given
2020-02-24 18:22:03,288 [DEBUG] conjure-up/charmed-kubernetes - __init__.py:27 - Showing dialog for exception: addCharm() takes 3 positional arguments but 4 were given
```

## Please provide the output of the following commands

```
$ which juju
/snap/bin/juju
$ juju version
2.7.2-bionic-amd64
$ which conjure-up
/snap/bin/conjure-up
$ conjure-up --version
conjure-up 2.6.10
$ which lxc
/usr/bin/lxc
$ /snap/bin/lxc config show
config:
  core.https_address: '[::]'
$ /snap/bin/lxc version
Client version: 3.21
Server version: 3.21
$ cat /etc/lsb-release
DISTRIB_ID=Ubuntu
DISTRIB_RELEASE=18.04
DISTRIB_CODENAME=bionic
DISTRIB_DESCRIPTION="Ubuntu 18.04.3 LTS"
```

Please attach tarball of **~/.cache/conjure-up**:

```
tar cvzf conjure-up.tar.gz ~/.cache/conjure-up
```
I'm not sure if I can do this due to the environment. If more info is needed please ask.

## Sosreport

Please attach a sosreport:

```
sudo apt install sosreport
sosreport
```

The resulting output file can be attached to this issue.

Again, not sure I can provide this for now.

## What Spell was Selected?

Charmed-kubernetes

## What provider (aws, maas, localhost, etc)?

Trying to use aws-gov but this run wouldn't let me use it, so chose localhost.

## Commands ran

Followed docs for installing conjure-up via snap. Tried to set up aws-gov cluster from an EC2 instance in the VPC with local juju but juju bootstrapping always failed with `ERROR failed to bootstrap model: no image metadata found`. Tried bootstrapping juju per docs. hello-juju worked fine (although took a long time to deploy). Tried `conjure-up charmed-kubernetes` with localhost as provider as aws-gov was not listed but this fails with the `addCharm` error.

**Comment by g-clef** [comment]
I'm seeing the same error when trying to install kubernetes on lxd. I think this is fixed by #1640 , but the most recent `--edge` release in snap is from Nov 09, so it does not include #1640 's fixes. If you all could bump the `--edge` release to include the Nov 29 changes, I'd be glad to try it out to see if this is fixed there.

**Comment by g-clef** [comment]
Update: I cloned the repo for conjure-up and made my own local snap out of it. I had to update one or two other libraries, but after that it did successfully deploy a kubernetes cluster to my lxd cluster. None of the optional installation options worked, however. 

Here's what I did: 
 * clone the repo, 
 * installed the snap tools to make your own snap, 
 * bumped the versions of `juju` and `juju-wait` up. (had to experiment to find a version that had both juju and juju-wait candidates)
 * build the conjure-up snap
 * `snap install` conjure-up from that local snapfile, 
 * use conjure-up as normal to install kubernetes.
